### PR TITLE
[react-router] Make SwitchProps children type more specific

### PR DESCRIPTION
**Pull request by MysteryBlokHed**
Related discussion: #58596

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test react-router`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://v5.reactrouter.com/web/api/Switch/children-node>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by MysteryBlokHed** [comment]
The failing tests seem to come from here, but I don't know why they're failing:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c1d75e3c1cca2959ad195bbeefd26ac8a878a2ce/types/react-router/test/Switch.tsx#L11-L19

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c1d75e3c1cca2959ad195bbeefd26ac8a878a2ce/types/react-router/test/examples-from-react-router-website/Ambiguous.tsx#L33-L37

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e33d3fe8b10acee209c6353529cf180b3cac25ac/types/react-router/test/examples-from-react-router-website/ModalGallery.tsx#L57-L61

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e33d3fe8b10acee209c6353529cf180b3cac25ac/types/react-router/test/examples-from-react-router-website/NoMatch.tsx#L21-L26

File: types/react-router/index.d.ts [hunk 1]
```diff
@@ -122,7 +122,7 @@ export interface StaticRouterProps {
 
 export class StaticRouter extends React.Component<StaticRouterProps, any> {}
 export interface SwitchProps {
-    children?: React.ReactNode | undefined;
+    children?: (Route | Redirect)[] | undefined;
```
Comment (eps1lon) [inline types/react-router/index.d.ts:125]: This is not the same as `<Route />` which is what's documented.

**Review by eps1lon — CHANGES_REQUESTED** [review]
Please add tests that show what children are no longer accepted and what children should be accepted.

File: types/react-router/index.d.ts (hunk 1 above)
Comment (MysteryBlokHed) [reply types/react-router/index.d.ts:125]: Which Route should I use instead?

Comment (eps1lon) [reply types/react-router/index.d.ts:125]: You probably want to use an element created from `Route` not the component itself.

**Pull request closed without merging**